web/src/message_edit.ts

```diff
@@ -854,10 +853,12 @@ export function end_message_row_edit($row: JQuery): void {
         message_lists.current.hide_edit_message($row);
         compose_call.abort_video_callbacks(message.id.toString());
     }
-    if ($row.find(".condensed").length !== 0) {
-        condense.show_message_expander($row);
-    } else {
-        condense.show_message_condenser($row);
+    if ($row.find(".could-be-condensed").length !== 0) {
+        if ($row.find(".condensed").length !== 0) {
+            condense.show_message_expander($row);
+        } else {
+            condense.show_message_condenser($row);
+        }
```

Comment:
Can you explain this change in logic?

Reply:
The check `$row.find(".could-be-condensed").length !== 0` was previously being used in `show_message_expander`/`show_message_condenser` to check if the message was actually something that required the `SHOW MORE`/`SHOW LESS` toggle buttons. Since we are now using the `show_message_expander`/`show_message_condenser` throughout the `condense.ts` as the common functions to toggle the buttons, I have moved this logic over to `message_edit.ts`.

We don't use the `$row.find(".could-be-condensed").length !== 0` check in the `show_message_expander`/`show_message_condenser` functions now, since they are now also used to toggle the `collapsed` state, which won't be having a `.could-be-condensed` attached to it.

Reply:
OK